### DISCUSSION TITLE
Multichannel support

### DISCUSSION THREAD

**Discussion by djniviro** [apohl79/audiogridder#955] [Ideas]
A feature request.

as I’m working in Nuendo in various immersive formats (up to 13.1 Auro3D), I cannot seem to find a way to make the Audiogridder plugin perform in a multichannel mode, even though up to 64 channels should be available.

a feature like Vienna Ensemble where you can select the channel configuration of the plugin (either 2.0, LCR, LCR+LFE, 5.1, 7.1, 7.1.2, 7.1.4, 9.1 Auro, 11.1 Auro and 13.1 Auro)

if this could happen, it would be amazing to see on the server side also a possibility to insert both the multichannel version of supporting plugins and multi-mono instances of plugins which don’t support multichannel configurations, according to the chosen channel configuration. I think this is one of the most important features ever which Cubase and Nuendo lack.

the current workaround is setting up multiple audiogridder plugin instances on a channelstrip and link these stereo instances to the according channels of the speakers. BUT, for some reason a lot of phase misalignment occurs then. Is Audiogridder completely phase alligned?

massive thanks for your comment in advance!

**Comment by apohl79** [comment]
Will have a look at this, thx!

**Reply by djniviro** [reply]
That would be awesome! Also, just as with Logic Pro X, it would be really handy if there would be an option to enable to have separate controls for the LFE channel in one group and the controls of all other channels in another group. Of course, that'll all be up to you!

**Reply by apohl79** [reply]
Unfortunately I can't download a VEP demo without buying something from them... ;-)

So I have some questions about how you think it should work.

Right now, when you insert an AG plugin, it will forward the bus/channel config set by the DAW to the server. If you insert a plugin into AG, the server tries to find a config with the maximum number of channels provided by the DAW. For example if I create a 3rd order ambisonics audio track in cubase, I get 18 channels (16 channels in & out + 2 sidechain channels). 

I updated the code a bit, that tries to find a matching layout for a plugin, but conceptually it stays like this.

What is the intended behaviour in your workflow?

**Reply by djniviro** [reply]
Thanks for getting back to me with this topic! As you mentioned, the AGridder DAW plugin indeed detects the channel lanes count. For example, when used on a 7.1 track, it shows 8+8, which is perfect so far. The main feature request is aimed towards the server section of the application.

As most plugins do come in both a stereo and mono option, it would be super handy, if a channel config larger than 2 channels gets chosen, the option should be provided loading in the selected plugin either as a mono, stereo or multi-mono with routing options. So let's say I want to use a plugin on my LFE channel, I should be able to load in a mono instance of the particular plugin and route it to the LFE channel (which in music is mostly ch.4). BUT the remaining 7 channels then have to be sample accurate latency compensated. 

Also, I don't know much about the VST design, but does it deliver the channel names as well? (Left, Right, Center, LFE,...) as depending on the DAW and selected mode, they intend to change (Left, Center, Right, Ls, Rs, Lss, Rss, LFE as for movies).

For me, it's mainly the multi-mono option which would be a life saver as Nuendo doesn't support it. To make it work in Nuendo I need to load in either a stereo or mono instance and route them to the according channels for now (sometimes I have to use up to 8 plugins stacked). Also when trying to adjust a parameter, all 8 have to be manually adjusted and it's kinda a pain in the ... :)

Is there something you'd want to know from VEP? Because from my experience, they do not support multi-mono however I could be wrong and a world could open up haha.

**Reply by apohl79** [reply]
Thx for this. Makes it much clearer now! I wanted to look at VEP to see how it works as you mentioned it. But your description is all I needed for now.

**Reply by djniviro** [reply]
You’re a hero! Please feel free to get in touch with me for alpha testing via hello@djniviro.com in case you need feedback.

oh btw, completely different issue, but when opening the plugins of the server app, everything is there, but some are lacking in the DAW plugin. Also when using an Agridder synth instance, not a single synth vst shows in the app. Is there a workaround for?

**Reply by apohl79** [reply]
> oh btw, completely different issue, but when opening the plugins of the server app, everything is there, but some are lacking in the DAW plugin. Also when using an Agridder synth instance, not a single synth vst shows in the app. Is there a workaround for?

try to disable the server filter in the plugin settings. the filter separates fx from instrument plugins, which is not super reliable.

**Reply by djniviro** [reply]
Will try in a bit if this works! Thanks for the quick reply!

The last thing I've experienced is that I need to have a remote desktop open before opening a project involving AudioGridder. If I don't, the whole DAW freezes as it's not able to get in the VNC connection. I suppose this has something to do with authorization. This being said, if you'd be able to give the VNC an option to login to the remote system, it'll also be possible having a multi-seat AudioGridder server as the dimensions of the VNC can be theoretically infinite. Have you considered developing a payware extended-feature version of AudioGridder?

Let's say a bigger studio with multiple rooms wants a centralised system to run plugins to make sure the local machines will not get clogged with VST processing. And each room wants to connect to the same central server, which currently is not possible as only one plugin can be opened at the same time (opened in the left top corner of the server screen). If you could allocate a maximum reserved VST plugin size (let's say 1024 x 1024) per connected client, you could add the same resolution to the VNC screen size when an extra client connects. The only thing to consider then are the controls of the plugins, as currently the mouse position of the AG plugin gets mirrored to the VNC window. If I'm correct it does recognise the generic parameters, so would it be possible to retrieve the position of every parameter and map it accordingly to the position of the VST plugin? This eliminates the use of a mouse on the AG server and makes it more accurate, as my Windows to Windows AG setup has a slight mouse offset at this moment.

Also when starting up the AG server, it does scan for plugins but not for its licenses until the plugin gets selected and opened. Would it be possible to virtually open all plugins to check for its licenses like a regular DAW such as Ableton and Nuendo does? Currently when starting up my AG server and then opening the Nuendo project files, I'll need to go back to the remote desktop to log in into the iLok cloud connection.

I'd pay for these features for sure! haha

**Reply by apohl79** [reply]
> The last thing I've experienced is that I need to have a remote desktop open before opening a project involving AudioGridder. If I don't, the whole DAW freezes as it's not able to get in the VNC connection. I suppose this has something to do with authorization. This being said, if you'd be able to give the VNC an option to login to the remote system, it'll also be possible having a multi-seat AudioGridder server as the dimensions of the VNC can be theoretically infinite.

That limit will probably stay, at least for now.. never say never. VNC or RDP (works with AG on Windows sometimes too, but showed issues in the past) need to open a desktop session first. I think there is ways to automate that process, like using auto login on macOS or Windows. Right now I don't see AG bundling VNC or messing with it, as it is not requirement really.

Regarding multi-user setups, I played with it only on macOS. You need to setup a separate user on the server for each AG instance you want to run with screen sharing, then log each user in in parallel (with VNC for instance). Now you have multiple parallel desktops. But that approach was running into issues with a few plugins like waves for example, which did not render the UIs always/reliably.

> Also when starting up the AG server, it does scan for plugins but not for its licenses until the plugin gets selected and opened. Would it be possible to virtually open all plugins to check for its licenses like a regular DAW such as Ableton and Nuendo does? Currently when starting up my AG server and then opening the Nuendo project files, I'll need to go back to the remote desktop to log in into the iLok cloud connection.

AG does only rescan changed plugins and load them right now, as the license check would require to create an instance of the plugin. This can become very slow, so this is not happening right now. If you would wipe the AG cache before staring the server and keep scanning at startup enabled, it would basically load every plugin when at start up time. There is no option for this right now, but you could script it.

> Have you considered developing a payware extended-feature version of AudioGridder?

Thought about it a long time ago and dropped that thought as I have no capacity for supporting it commercially.

Thanks for your input!
